src/isomorphism.rs

```diff
@@ -66,7 +68,11 @@ where
             let c_index = graph.from_index(c);
             new_graph.add_edge(p_index, c_index, edge_w.clone_ref(py));
         }
-
+        if mapping.is_some() {
+            for (new_index, old_value) in id_map.iter().enumerate() {
+                mapping.as_mut().unwrap().insert(*old_value, new_index);
```

Comment:
```suggestion
            for (old_value, new_index) in id_map.iter().enumerate() {
                mapping.as_mut().unwrap().insert(*new_index, old_value);
```
swap the names of these variables to reflect the semantics of `id_map`.